### PR TITLE
[Function-NoOp] Integration test to verify skipping deploy is WAI

### PR DESCRIPTION
**Pull request by TheIronDev**
### Description

Integration test to verify skipping deploy is WAI

### Scenarios Tested

- [x] NoOp Function Deploys are skipped
- [ ] Non-NoOp Function Deploys are skipped

### CODE REVIEW

File: scripts/functions-deploy-tests/tests.ts [hunk 1]
```diff
@@ -260,6 +263,83 @@ describe("firebase deploy", function (this) {
...
+        schedule: "every 30 minutes",
+        retryCount: 3,
+        minBackoffSeconds: 42,
+        maxRetrySeconds: 42,
+        maxDoublings: 42,
+        maxBackoffSeconds: 42,
+      },
+    };
+
+    previews.skipdeployingnoopfunctions = true;
```
Comment (colerogers) [inline scripts/functions-deploy-tests/tests.ts:335]: q: do we plan on removing this preview flag? If so, we should put a TODO here to yank it

**Review by colerogers — COMMENTED** [review]
let me know when this is out of draft state and I'll do another pass

File: scripts/functions-deploy-tests/tests.ts [hunk 2]
```diff
@@ -260,6 +263,83 @@ describe("firebase deploy", function (this) {
     }
   });
 
+  it("skips duplicate deploys functions with runtime options", async () => {
+    const opts: Opts = {
```
Comment (taeold) [inline scripts/functions-deploy-tests/tests.ts:267]: It looks like this configuration must match the configuration set in the previous test case in order for the test case to work as intended.

Can we make that coupling explicit? e.g. pull out configuration as a test variable.

File: scripts/functions-deploy-tests/tests.ts (hunk 1 above)
Comment (TheIronDev) [reply scripts/functions-deploy-tests/tests.ts:335]: Updated!

File: scripts/functions-deploy-tests/tests.ts (hunk 2 above)
Comment (TheIronDev) [reply scripts/functions-deploy-tests/tests.ts:267]: They don't need to match the previous one, I was just copy/pasting. I updated the opts to not be the giant list (to reduce distractions from the purpose of the test.)

**Comment by TheIronDev** [comment]
@taeold Any idea why test returns this:

```
- YAMLException incomplete explicit mapping pair; a key node is missed; or followed by a non-tabulated empty line at line 1, column 62:
     ... rom function source: SyntaxError: Unexpected token '?'
```

Might be an issue with the project I am using....

**Review by taeold — APPROVED** [review]

**Review by colerogers — APPROVED** [review]

**Pull request merged**